`scan_csv` with `skip_rows > 0` followed by a `slice` starts reading values from the first row of the file

#### Are you using Python or Rust?

Python

#### What version of polars are you using?

0.12.21 (48ea1ed03)

#### What operating system are you using polars on?

Fedora Linux 35

#### Describe your bug.

Thank you again for the new `skip_rows` options! It looks like there's a bug when mixed with slices though. If you use `pl.scan_csv` with `skip_rows` greater than 0 to read a file, call `slice` (or `head`), and then `collect`, the `slice` offset will be calculated from the first row of the *file*, not after `skip_rows`. In cases where the first few lines are metadata, this will result in an error.

#### What are the steps to reproduce the behavior?

Here are some tests to add to `test_csv_schema_offset` in `test_io.py` that show the problem. The first one passes but not the second.

```python
    df = pl.scan_csv(foods_csv, skip_rows=4).collect()
    assert df[0,0] == "seafood"
    assert df.shape == (23, 4)

    df = pl.scan_csv(foods_csv, skip_rows=4).head(1).collect()
    assert df[0,0] == "seafood"
    assert df.shape == (1, 4)
```

#### What is the actual behavior?

```
        df = pl.scan_csv(foods_csv, skip_rows=4).head(1).collect()
>       assert df[0,0] == "seafood"
E       AssertionError: assert 'vegetables' == 'seafood'
E         - seafood
E         + vegetables

tests/test_io.py:585: AssertionError
```

#### What is the expected behavior?

The tests should pass.

--- a/polars/polars-lazy/src/frame/mod.rs
+++ b/polars/polars-lazy/src/frame/mod.rs
@@ -938,7 +938,7 @@ impl LazyFrame {
         self.slice(-1, 1)
     }
 
-    /// Get the n last rows
+    /// Get the last `n` rows
     pub fn tail(self, n: u32) -> LazyFrame {
         let neg_tail = -(n as i64);
         self.slice(neg_tail, n)

--- a/polars/polars-lazy/src/logical_plan/optimizer/slice_pushdown.rs
+++ b/polars/polars-lazy/src/logical_plan/optimizer/slice_pushdown.rs
@@ -155,7 +155,7 @@ impl SlicePushDown {
                 aggregate,
             }, Some(state)) if state.offset >= 0 && predicate.is_none() => {
                 let mut options = options;
-                options.skip_rows = state.offset as usize;
+                options.skip_rows += state.offset as usize;
                 options.n_rows = Some(state.len as usize);
 
                 let lp = CsvScan {

--- a/polars/polars-lazy/src/tests/io.rs
+++ b/polars/polars-lazy/src/tests/io.rs
@@ -316,3 +316,15 @@ fn test_slice_filter() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn skip_rows_and_slice() -> Result<()> {
+    let out = LazyCsvReader::new(FOODS_CSV.to_string())
+        .with_skip_rows(4)
+        .finish()?
+        .limit(1)
+        .collect()?;
+    assert_eq!(out.column("fruit")?.get(0), AnyValue::Utf8("seafood"));
+    assert_eq!(out.shape(), (1, 4));
+    Ok(())
+}

--- a/polars/polars-lazy/src/tests/projection_queries.rs
+++ b/polars/polars-lazy/src/tests/projection_queries.rs
@@ -35,6 +35,7 @@ fn test_join_suffix_and_drop() -> Result<()> {
 }
 
 #[test]
+#[cfg(feature = "cross_join")]
 fn test_cross_join_pd() -> Result<()> {
     let food = df![
         "name"=> ["Omelette", "Fried Egg"],

